packages/next/src/client/dev/error-overlay/websocket.ts

```diff
@@ -39,11 +41,17 @@ export function connectHMR(options: {
       lastActivity = Date.now()
     }
 
-    function handleMessage(event: any) {
+    function handleMessage(event: MessageEvent<string>) {
       lastActivity = Date.now()
 
+      // webpack's heartbeat event.
+      if (event.data === '\uD83D\uDC93') {
+        return
+      }
+
+      const msg = JSON.parse(event.data)
       eventCallbacks.forEach((cb) => {
-        cb(event)
+        cb(msg)
       })
```

Comment:
WebSocket listeners (the things passed to `addMessageListener`) will now receive a parsed JSON object instead of the string that they would need to parse themselves. Small micro optimization.